AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -81,6 +75,36 @@ - (void)channel:(id<MSChannelProtocol>)__unused channel prepareLog:(id<MSLog>)lo
   }
 }
 
+- (void)channel:(id<MSChannelProtocol>)channel
+     didPrepareLog:(id<MSLog>)log
+    withInternalId:(NSString *)__unused internalId {
+  if (![self shouldSendLogToOneCollector:log] || ![channel conformsToProtocol:@protocol(MSChannelUnitProtocol)]) {
+    return;
+  }
+  id<MSChannelUnitProtocol> channelUnit = (id<MSChannelUnitProtocol>)channel;
+  NSString *groupId = channelUnit.configuration.groupId;
+  id<MSChannelUnitProtocol> oneCollectorChannelUnit = [self.oneCollectorChannels objectForKey:groupId];
+  if (!oneCollectorChannelUnit) {
+    return;
+  }
+  id<MSLogConversion> logConversion = (id<MSLogConversion>)log;
+  NSArray<MSCommonSchemaLog *> *commonSchemaLogs = [logConversion toCommonSchemaLogs];
+  for (MSCommonSchemaLog *commonSchemaLog in commonSchemaLogs) {
```

Comment:
`toCommonSchemaLogs` could return `nil`. You don't need `nil` check before iteration?

Reply:
No for loop will simply ignore nil arrays.